src/parser/mod.rs

```diff
@@ -3479,18 +3473,22 @@ impl<'a> Parser<'a> {
     }
 
     /// Parse a comma-separated list of 0+ items accepted by `F`
-    pub fn parse_comma_separated0<T, F>(&mut self, f: F) -> Result<Vec<T>, ParserError>
+    /// * `trailing_commas` - support trailing_commas or not
+    /// * `end_token` - expected end token for the closure (e.g. [Token::RParen], [Token::RBrace] ...)
+    pub fn parse_comma_separated0<T, F>(
+        &mut self,
+        f: F,
+        trailing_commas: bool,
```

Comment:
I also extract this argument because this option isn't shared by different end_token in a dialect. For example, DuckDB supports the trailing_commas for the parenthesis.
```
select * from t where a in (1,2,);
```
However, it doesn't support the trailing_commas for the braces.
```
D select {,};
Parser Error: syntax error at or near ","
LINE 1: select {,};
```

Reply:
hmm I'm not too sure if pulling out the trailing_comma option is worth it/desirable - the current behavior isn't specified on the dialect level, but rather on the parser itself, to be overridden by the user, and I wonder if it'll get confusing to have to figure out where each dialect supports or doesnt support trailing commas. Would it be okay to keep the behavior as it was before? (i.e. always guarding internally on `self.options.trailing_comma`)

Reply:
> hmm I'm not too sure if pulling out the trailing_comma option is worth it/desirable - the current behavior isn't specified on the dialect level, but rather on the parser itself, to be overridden by the user, and I wonder if it'll get confusing to have to figure out where each dialect supports or doesnt support trailing commas. Would it be okay to keep the behavior as it was before? (i.e. always guarding internally on `self.options.trailing_comma`)

I see. I tried more cases for DuckDB, I found the trailing_comma behaviors are shared.
```sql
D select * from t where a in (1,2,);
┌────────┐
│   a    │
│ int32  │
├────────┤
│ 0 rows │
└────────┘
D select {'a':1,};
┌──────────────────────────┐
│ main.struct_pack(a := 1) │
│    struct(a integer)     │
├──────────────────────────┤
│ {'a': 1}                 │
└──────────────────────────┘
D select [1,2,];
┌───────────────────────┐
│ main.list_value(1, 2) │
│        int32[]        │
├───────────────────────┤
│ [1, 2]                │
└───────────────────────┘
```
I think we can use `self.options.trailing_comma` internally.